### PR TITLE
Space out refresh-mcollective-metadata cron task

### PR DESCRIPTION
**Pull request by pkrul**
refresh-mcollective-metadata is being run at four fixed moments within each hour. In certain  configurations with hundreds of virtual machines, all managed by puppet, this simultaneous start of  refresh-mcollective-metadata is causing a large CPU steal (ready) time, from seconds to minutes, during which the hypervisors need to deal with the large sudden demand of CPU cycles coming from virtual machines that are agressively competing for CPU resources. To prevent these outages, the refresh-mcollective-metadata cron job should be spaced out over various moments in time, and started less often.

Hypervisor with simultaneous starts of refresh-mcollective-metadata via cron:
cpu-ready-without-cron-patch

Hypervisor with spaced out start of the refresh-mcollective-metadata cron jobs (rolling out patch on  virtual machines started at 15:20)
cpu-ready-with-cron-patch

### CODE REVIEW

File: manifests/server/config/factsource/yaml.pp [hunk 1]
```diff
@@ -7,6 +7,9 @@
   $excluded_facts      = $mcollective::excluded_facts
   $yaml_fact_path_real = $mcollective::yaml_fact_path_real
 
+  $cron_minute_value   = fqdn_rand(60, ${::macaddress})
+  $cron_hour_value     = '0-23/2'
```
Comment (nibalizer) [inline manifests/server/config/factsource/yaml.pp]: I think this assumes a lot about the cron capabilities of the host. What if we did the below? Then fired on every hour.

$cron_minute_offset = fqdn_rand(15, ${::macaddress})
$cron_minutes = [ $cron_minute_offset, $cron_minute_offset + 15, $cron_minute_offset + 30, $cron_minute_offset + 45 ]

Comment (igalic) [inline manifests/server/config/factsource/yaml.pp]: +1

**Comment by pkrul** [comment]
@nibalizer Regarding cron capabilities, it's BSD compatible. That said, your solution fits mcollective better, so +1 to that. Thanks!

**Comment by igalic** [comment]
@pkrul it may be BSD compatible, but it won't work on Solaris, or AIX, or any other weird os ;)

**Comment by pkrul** [comment]
@igalic ah, not AT&T compatible ;)

**Comment by ffrank** [comment]
But is 2015 truly the year of WeirdOS on the desktop?

**Comment by nibalizer** [comment]
@pkrul so can you update the pr with my suggestion?

**Comment by pkrul** [comment]
@nibalizer done. is this acceptable for you?

**Comment by nibalizer** [comment]
Looks good.

**Comment by igalic** [comment]
[tests failing with](https://travis-ci.org/puppet-community/puppet-mcollective/jobs/58135555#L144)
```
Could not parse for environment *root*: Could not match ${::macaddress}) at /home/travis/build/puppet-community/puppet-mcollective/manifests/server/config/factsource/yaml.pp:10
rake aborted!
/home/travis/.rvm/gems/ruby-1.9.3-p551/gems/puppet-syntax-2.0.0/lib/puppet-syntax/tasks/puppet-syntax.rb:41:in `block (2 levels) in initialize'
/home/travis/.rvm/gems/ruby-1.9.3-p551/gems/puppetlabs_spec_helper-0.10.1/lib/puppetlabs_spec_helper/rake_tasks.rb:237:in `block in <top (required)>'
/home/travis/.rvm/gems/ruby-1.9.3-p551/bin/ruby_executable_hooks:15:in `eval'
/home/travis/.rvm/gems/ruby-1.9.3-p551/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => syntax => syntax:manifests
(See full trace by running task with --trace)
The command "bundle exec rake validate && bundle exec rake lint && bundle exec rake spec SPEC_OPTS='--format documentation'" exited with 1.
```

anyone got an idea why that is?

File: manifests/server/config/factsource/yaml.pp [hunk 2]
```diff
@@ -7,6 +7,10 @@
   $excluded_facts      = $mcollective::excluded_facts
   $yaml_fact_path_real = $mcollective::yaml_fact_path_real
 
+  $cron_minute_offset  = fqdn_rand(15, ${::macaddress})
```
Comment (juniorsysadmin) [inline manifests/server/config/factsource/yaml.pp]: Er, this is probably better:
```
$cron_minute_offset  = fqdn_rand(15, $::macaddress)
```

Comment (igalic) [inline manifests/server/config/factsource/yaml.pp]: gosh, i didn't even see that

Comment (nibalizer) [inline manifests/server/config/factsource/yaml.pp]: Hehe

**Comment by igalic** [comment]
see https://github.com/puppet-community/puppet-mcollective/pull/206 for other lint failures.

**Comment by igalic** [comment]
@pkrul can you please rebase this against master again? should work now in travis!

**Comment by nibalizer** [comment]
@pkrul see @igalic last?

**Comment by jyaworski** [comment]
This looks like it's no longer needed. Please re-open and rebase if it is.

**Pull request closed without merging**